stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/processor/LocalGovActionStateProcessor.java

```diff
@@ -0,0 +1,60 @@
+package com.bloxbean.cardano.yaci.store.governance.processor;
+
+import com.bloxbean.cardano.yaci.store.events.EpochChangeEvent;
+import com.bloxbean.cardano.yaci.store.events.RollbackEvent;
+import com.bloxbean.cardano.yaci.store.governance.service.LocalGovStateService;
+import com.bloxbean.cardano.yaci.store.governance.storage.impl.repository.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.concurrent.TimeUnit;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+@ConditionalOnBean(LocalGovStateService.class)
+public class LocalGovActionStateProcessor {
+    private final LocalGovStateService localGovStateService;
+    private final LocalGovActionProposalStatusRepository localGovActionProposalStatusRepository;
+    private final LocalConstitutionRepository localConstitutionRepository;
+    private final LocalCommitteeMemberRepository localCommitteeMemberRepository;
+    private final LocalCommitteeRepository localCommitteeRepository;
+    private final LocalTreasuryWithdrawalRepository localTreasuryWithdrawalRepository;
+
+    @EventListener
+    @Transactional
+    public void handleEpochChangeEvent(EpochChangeEvent epochChangeEvent) {
+        if (!epochChangeEvent.getEventMetadata().isSyncMode())
```

Comment:
What happens when we reach sync mode = true during initial sync, but we are still in the middle of the epoch ?

Should we somehow trigger this fetch once the sync reaches tip or close to tip even if the next epoch is not there yet ?